lib/service.rb

```diff
@@ -101,7 +113,7 @@ def pid?
     def error?
       return false if pid?
 
-      exit_code.blank? || exit_code.nonzero?
+      !(exit_code.present? && exit_code.nonzero?)
```

Comment:
Is this right? Currently `#error?` is `false` whenever `exit_code.nonzero?` is `true`.

Reply:
Hmm, I'm not sure. I think I did this because it being empty isn't a guarantee that it's an error.

I think it should only be an error if the value is filled AND non-zero.